src/swap/SwapScreen.tsx

```diff
@@ -481,8 +481,14 @@ export function SwapScreen({ route }: Props) {
         <PreparedTransactionsReviewBottomSheet
           forwardedRef={preparedTransactionsReviewBottomSheetRef}
           preparedTransactions={exchangeRate.preparedTransactions}
-          onAcceptDecreaseSwapAmountForGas={({ decreasedSwapAmount }) => {
-            handleChangeAmount(updatedField)(decreasedSwapAmount.toString())
+          onAcceptDecreaseSwapAmountForGas={({ decreasedSpendAmount }) => {
+            handleChangeAmount(updatedField)(
+              // ensure units are for the asset whose amount is being selected by the user
```

Comment:
I moved this out of `prepareTransactions` since it is specific to sends, and I thought it made the helper easier to reason about for its outputs to have consistent units